docx/header.py

```diff
@@ -34,6 +35,19 @@ def is_linked_to_previous(self):
             return True
         return False
 
+    @lazyproperty
+    def body(self):
+        """
+        BlockItemContainer instance with contents of Header
+        """
+        headerRef = self._sectPr.get_headerReference_of_type(self._type)
+        if headerRef is None:
+            return None
+
+        # should always have an r:id or else it's a really weird headerRef...
+        rId = headerRef.attrib['{%s}id' % NS.OFC_RELATIONSHIPS]
+        return self.part.related_header_body(rId)
```

Comment:
Ok, basically it looks like you haven't implemented the oxml layer yet. You shouldn't need any lxml code in this (API)  layer. So sectPr is a CT_SectPr type and is defined in docx.oxml.section. It has `w:headerReference` defined as a ZeroOrMore child, but the `w:headerReference` tag name is not registered in `docx/oxml/__init__.py`, so all headerReference elements get loaded as a regular lxml class with no custom behaviors.

In order to register it, you need a custom element class, CT_whatever, probably in docx/oxml/header.py. The class name is spelled the same as the XML complex type name. Then you can define attributes on it like id and rId.

The attribute definition would be:

```python
rId = OptionalAttribute('r:id', ST_RelationshipId)
```

Then getting the rId value is just `rId = headerReference.rId`. Note the convention is to spell the element variable name exactly as the tag name (less the namespace prefix), so headerReference rather than headerRef.